web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -74,18 +74,18 @@ export class DataStoreService {
       .then(() => projectId);
   }
 
-  updateProjectLayer(projectId: string, layers: string) {
-    const updatedLayers = JSON.parse(layers);
+  // TODO: Define return types for methods in this class
+  updateProjectLayer(projectId: string, layers: {}) {
     return new Promise((resolve, reject) => {
       this.db
         .collection('projects')
         .doc(projectId)
-        .set({ layers: updatedLayers }, { merge: true })
+        .update(layers)
```

Comment:
This method should remove `id` from the Layer before writing it, since unlike in our application model, the database "Layer" object doesn't contain `id` as a member.